File: rust/src/models/rate_plan_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::term_rate::TermRate;

/// Recurring rate fee (e.g., monthly subscription)
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct RatePlanFee {
    pub rates: Vec<TermRate>,
}

impl RatePlanFee {
    pub fn new(rates: Vec<TermRate>) -> Self {
        Self { rates }
    }
}

File: typescript/src/models/ratePlanFee.ts
// this file is @generated
import { type TermRate, TermRateSerializer } from "./termRate";
/** Recurring rate fee (e.g., monthly subscription) */
export interface RatePlanFee {
  rates: TermRate[];
}

export const RatePlanFeeSerializer = {
  _fromJsonObject(object: any): RatePlanFee {
    return {
      rates: object["rates"].map((item: any) => TermRateSerializer._fromJsonObject(item)),
    };
  },

  _toJsonObject(self: RatePlanFee): any {
    return {
      rates: self.rates.map((item: any) => TermRateSerializer._toJsonObject(item)),
    };
  },
};
